twenty-first/src/shared_math/b_field_element.rs

```diff
@@ -160,55 +160,31 @@ impl BFieldElement {
         acc
     }
 
-    pub fn legendre_symbol(&self) -> i8 {
-        let elem = self.mod_pow((Self::QUOTIENT - 1) as u64 / 2).0;
-
-        // Ugly hack to force a result in {-1,0,1}
-        if elem == Self::QUOTIENT - 1 {
-            -1
-        } else if elem == 0 {
-            0
-        } else {
-            1
-        }
-    }
-
     /// Convert a `BFieldElement` from a byte slice.
     pub fn from_ne_bytes(bytes: &[u8]) -> BFieldElement {
         let mut bytes_copied: [u8; 8] = [0; 8];
         bytes_copied.copy_from_slice(bytes);
         BFieldElement::new(u64::from_ne_bytes(bytes_copied))
     }
 
+    /// Montgomery reduction
     #[inline(always)]
-    fn mod_reduce(x: u128) -> u64 {
-        // Copied from (MIT licensed):
-        // https://github.com/anonauthorsub/asiaccs_2021_440/blob/5141f349e5915ab750208c0ab1b5f7be95adaeac/math/src/field/f64/mod.rs#L551
-        // assume x consists of four 32-bit values: a, b, c, d such that a contains 32 least
-        // significant bits and d contains 32 most significant bits. we break x into corresponding
-        // values as shown below
-        let ab = x as u64;
-        let cd = (x >> 64) as u64;
-        let c = (cd as u32) as u64;
-        let d = cd >> 32;
-
-        // compute ab - d; because d may be greater than ab we need to handle potential underflow
-        let (tmp0, under) = ab.overflowing_sub(d);
-        let tmp1 = tmp0.wrapping_sub(Self::LOWER_MASK * (under as u64));
-
-        // compute c * 2^32 - c; this is guaranteed not to underflow
-        let tmp2 = (c << 32) - c;
-
-        // add temp values and return the result; because each of the temp may be up to 64 bits,
-        // we need to handle potential overflow
-        let (result, over) = tmp1.overflowing_add(tmp2);
-        result.wrapping_add(Self::LOWER_MASK * (over as u64))
+    const fn mod_reduce(x: u128) -> u64 {
```

Comment:
Since this method does not compute modular reduction, I would suggest not calling it that.

The [source material](https://eprint.iacr.org/2022/274.pdf) calls it `montyred`, which is presumably short for "Montgomery reduction", which is distinct from modular reduction modulo $p$.